### DISCUSSION TITLE
How close all the routes in intercepting routes?

### DISCUSSION THREAD

**Discussion by Neo-Ciber94** [vercel/next.js#59466] [App Router]
Look at this example code: https://stackblitz.com/edit/stackblitz-starters-xaeywm

If I had multiple routes intercepted:

- @modal
- @modal/(.)photos
- @modal/.)photos/[id]

In the documentation say to use `router.back()` but you are just navigating back between modals, how to close all modals, so going to the start of the route.

**Comment by davidguy3237** [comment]
I'm actually curious about this too.
I'm doing the parallel/intercepting route method to turn my login and sign up page into modals. There's a link to switch between login and signup.

My current method to close the modal is to use `router.back()` like what is suggested in the docs, but if you're switching between multiple intercepted routes and then close the modal, then it has to go through every previous modal intercept before the modal finally closes.

Here's a short video showing it.

https://github.com/vercel/next.js/assets/112038666/839ac586-4580-4d2b-bdea-7249290e42c6

I could force a redirect forward with like `router.push()` or `<Link>`, but since the login button is in a navbar that appears on multiple pages, that doesn't seem like a good user experience if I just forced a redirect to the home page or somewhere else. Also I'm not sure how you would get access to the original URL from before the modal opened.

I'm wondering if this just isn't a good use case for this feature.

**Reply by gsdev01** [reply]
any update on this ?

**Reply by gastoncorti** [reply]
When navigating between modals use replace instead of push.

**Comment by dcapitator** [comment]
I am also having that ux issue for login/signup .. 

router.back() is navigating between modals so if i click signup and then signin x5 then closing the modal would navigate back 5 times each modal !! @icyJoseph @samx

**Comment by dcapitator** [comment]
what if i am in a parallel route already for example /signin and i click signup , how to prevent intercepting signup route/modal ??

do we have conditional route intercepting ?

**Comment by younessaita** [comment]
any updates?

**Reply by younessaita** [reply]
nevermind. i used router.replace instead of push so the url don't get pushed to the history stack. keep in mind this will need some implementation to check if the link is displaying inside a modal or in a page, because if you are inside the page itself, you actually need router push so you can get back to it. Although, inside the modal you always want your link to be replaced not pushed to the history.

**Comment by heron2014** [comment]
This answer helped me https://github.com/vercel/next.js/discussions/50284#discussioncomment-9094733
